composeApp/src/androidMain/kotlin/io/github/couchtracker/db/user/UserDbResult.kt

```diff
@@ -83,3 +93,21 @@ fun <T, R> UserDbResult<T>.map(block: (T) -> R): UserDbResult<R> {
         is UserDbResult.FileError.UriCannotBeOpened -> this
     }
 }
+
+/**
+ * Calls [block] whenever [this] is not a [UserDbResult.Completed.Success].
+ *
+ * Being inline, allows idiomatic code such as:
+ * ```kotlin
+ * value.onError { return it }
+ * ```
+ */
+inline fun <T> UserDbResult<T>.onError(block: (UserDbResult<Nothing>) -> Unit) {
+    when (this) {
+        is UserDbResult.Completed.Success -> {}
+        is UserDbResult.Completed.Error -> block(this)
+        is UserDbResult.FileError.ContentProviderFailure -> block(this)
+        is UserDbResult.FileError.InvalidDatabase -> block(this)
+        is UserDbResult.FileError.UriCannotBeOpened -> block(this)
```

Comment:
These three could be merged into a simpler `is UserDbResult.FileError`

Reply:
Good point, but you approved and it auto merged